BDArmory/Control/MissileFire.cs

```diff
@@ -773,11 +777,21 @@ public void ToggleGuardMode()
                         {
                             if (rd.Current != null || rd.Current.canLock)
                             {
-                                rd.Current.EnableRadar();
-                                float scanSpeed = rd.Current.directionalFieldOfView / rd.Current.scanRotationSpeed * 2;
-                                if (GpsUpdateMax > 0 && scanSpeed < GpsUpdateMax) GpsUpdateMax = scanSpeed;
+                                if (rd.Current.sonarMode == ModuleRadar.SonarModes.None)
+                                {
+                                    rd.Current.EnableRadar();
+                                    float scanSpeed = rd.Current.directionalFieldOfView / rd.Current.scanRotationSpeed * 2;
+                                    if (GpsUpdateMax > 0 && scanSpeed < GpsUpdateMax) GpsUpdateMax = scanSpeed;
+                                    _radarsEnabled = true;
+                                }
+                                else
+                                {
+                                    rd.Current.EnableRadar();
+                                    float scanSpeed = rd.Current.directionalFieldOfView / rd.Current.scanRotationSpeed * 2;
+                                    if (GpsUpdateMax > 0 && scanSpeed < GpsUpdateMax) GpsUpdateMax = scanSpeed;
+                                    _sonarsEnabled = true;
+                                }
```

Comment:
Do we want both Active and Passive sonars activated at guard, or just Passive, with Active toggled when needed for targeting?